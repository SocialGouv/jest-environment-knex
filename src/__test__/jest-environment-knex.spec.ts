//

import "knex";
import { join } from "path";
const { databaseName, knex } = global;

beforeAll(async () => {
  await knex.migrate.latest({
    directory: join(__dirname, "../../dist/migrations"),
    loadExtensions: ["js"]
  });
});

test("should expose an 'databaseName' on global", () => {
  expect(databaseName).toMatch(/jest_environment_knex_\w{16}/);
});

test("should read the lorem db", async () => {
  expect(await knex.select().from("knex_migrations_lock")).toMatchSnapshot();
});
