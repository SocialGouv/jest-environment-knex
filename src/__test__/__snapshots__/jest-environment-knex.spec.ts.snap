// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should read the lorem db 1`] = `
Array [
  Object {
    "index": 1,
    "is_locked": 0,
  },
]
`;
